plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -859,7 +864,12 @@ else if (strings.size() != 2) {
             String schema = strings.get(0);
```

Comment:
Looks like `getMaterializedViewToken` above can also fail if materialized view deleted concurrently. Can it be so?

Reply:
I am not aware of where it could fail since `catalog.loadTable` throws `TableNotFoundException`, and no null value is returned by this method. This method also does not call `getTableHandle` which returns null.

Reply:
Ideally, in getMaterializedViewToken, the fully qualified <catalog name>.<schema name>.<table name> format is an invariant that is enforced by `Table` in apache iceberg library when calling:

```
String dependsOnTables = icebergTable.currentSnapshot().summary().getOrDefault(DEPENDS_ON_TABLES, "");
```

and just before splitting it and placing the values in a map.


Regardless, `else if (strings.size() != 2) {` ensures that index 0 and 1 are accessible.

Reply:
Actually good catch, that could be race condition. After looking at the implementation of the method, there does not seem to be anywhere that returns null or improperly handles null:

https://github.com/trinodb/trino/blob/9d73101322fe1bd22b315453a208350fb5736ad1/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java#L321

https://github.com/trinodb/trino/blob/9d73101322fe1bd22b315453a208350fb5736ad1/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L146

However, `TableNotFoundException` could be thrown. If the MV is deleted right before `catalog.loadTable(session, storageTableName)`, then I'd imagine its storage table would be removed atomically or sometime after?

We can add also catch `TableNotFoundException` in `MaterializedViewSystemTable`.